HLTrigger/Muon/plugins/HLTMuonL1TFilter.cc

```diff
@@ -110,7 +113,13 @@ bool HLTMuonL1TFilter::hltFilter(edm::Event& iEvent,
       MuonRef muon(allMuons, distance(allMuons->begin(allMuons->getFirstBX()), it));
 
       // Only select muons that were selected in the previous level
-      if (find(prevMuons.begin(), prevMuons.end(), muon) == prevMuons.end())
+      bool matchPrevL1 = false;
+      int prevSize = prevMuons.size();
+      for (int it2 = 0; it2 < prevSize; it2++) {
+        if (deltaR2(muon->eta(), muon->phi(), prevMuons[it2]->eta(), prevMuons[it2]->phi()) < maxDR_)
```

Comment:
General comment: there is some ambiguity in "deltaR vs deltaR2" (the parameter name suggests it's for DeltaR, while the actual cut is on DeltaR2). As done already here, the actual cut should be on DeltaR^2. My suggestion to mimic how this is handled [here](https://github.com/cms-sw/cmssw/blob/cf1af5f9671cb22fd9245813710e8ccdec100a67/HLTrigger/JetMET/interface/HLTPFJetsMatchedToFilteredJetsProducer.h#L48), reading the "maxDeltaR" value from the configuration, checking that it is positive, and then using the maxDeltaR2 value for the cut. Please also change `maxDR` to `maxDeltaR` to improve clarity.

Reply:
Ok, I have implemented it according to the suggestion.